src/main/java/lotto/domain/LottoMachine.java

```diff
@@ -0,0 +1,146 @@
+                collect(Collectors.toList());
+        return new LottoAutomaticTickets(lottoAutomaticTicket);
+    }
+
+    public List<LottoMatchStatus> lottoManualTicketsDiscriminator(
+            LottoManualTickets lottoManualTickets,
+            LastWeekWinningLotto lastWeekWinningLotto,
+            LastWeekWinningBonusBall lastWeekWinningBonusBall) {
+        List<LottoMatchStatus> lottoMatchStatuses = new ArrayList<>();
+
+        for (LottoManualTicket lottoManualTicket : lottoManualTickets.getLottoManualTickets()) {
+            int matchedLottoManualTicketCount = calculateMatchedLottoManualTicket(lottoManualTicket, lastWeekWinningLotto);
+            Boolean isMatchedBonusBall = calculateMatchedBonusBallManualTicket(lottoManualTicket, lastWeekWinningBonusBall);
+            LottoMatchStatus lottoMatchStatus = new LottoMatchStatus(matchedLottoManualTicketCount, isMatchedBonusBall);
+            lottoMatchStatuses.add(lottoMatchStatus);
+        }
+        return lottoMatchStatuses;
+    }
+
+    public List<LottoMatchStatus> lottoAutomaticTicketsDiscriminator(
+            LottoAutomaticTickets lottoAutomaticTickets,
+            LastWeekWinningLotto lastWeekWinningLotto,
+            LastWeekWinningBonusBall lastWeekWinningBonusBall) {
+        List<LottoMatchStatus> lottoMatchStatuses = new ArrayList<>();
+
+        for (LottoAutomaticTicket lottoAutomaticTicket : lottoAutomaticTickets.getLottoAutomaticTickets()) {
+            int matchedLottoAutomaticTicketCount = calculateMatchedLottoAutomaticTicket(lottoAutomaticTicket, lastWeekWinningLotto);
+            Boolean isMatchedBonusBall = calculateMatchedBonusBallAutomaticTicket(lottoAutomaticTicket, lastWeekWinningBonusBall);
+            LottoMatchStatus lottoMatchStatus = new LottoMatchStatus(matchedLottoAutomaticTicketCount, isMatchedBonusBall);
+            lottoMatchStatuses.add(lottoMatchStatus);
+        }
+        return lottoMatchStatuses;
+    }
+
+    private Boolean calculateMatchedBonusBallAutomaticTicket(
+            LottoAutomaticTicket lottoAutomaticTicket,
+            LastWeekWinningBonusBall lastWeekWinningBonusBall) {
+
+        List<Integer> lottoTicket = lottoAutomaticTicket.getLotto();
+        int WinningBonusBalls = lastWeekWinningBonusBall.getLastWeekWinningBonusBall();
+        return lottoTicket.contains(WinningBonusBalls);
+    }
+
+    private int calculateMatchedLottoAutomaticTicket(
+            LottoAutomaticTicket lottoAutomaticTicket,
+            LastWeekWinningLotto lastWeekWinningLotto) {
+
+        List<Integer> lottoTicket = lottoAutomaticTicket.getLotto();
+        List<Integer> lastWeekWinningLottoTicket = lastWeekWinningLotto.getLotto();
+        int matchedCount = 0;
+        for (int lottoNumber = 0; lottoNumber < LOTTO_SIZE; lottoNumber++) {
+            Boolean isMatchLottoNumber = lottoTicket.contains(lastWeekWinningLottoTicket.get(lottoNumber));
+            matchedCount += countMatchedLottoNumber(isMatchLottoNumber);
```

Comment:
```suggestion
            if(lottoTicket.contains(lastWeekWinningLottoTicket.get(lottoNumber)) {
                    matchCount++;
            }
            continue;
```
이렇게 하고 `lottoTicket.contains(lastWeekWinningLottoTicket.get(lottoNumber)`를 적절한 이름으로 추출하는 것이 더 나아보여요